PSReadLine/ReadLine.cs

```diff
@@ -87,9 +92,33 @@ private void ReadOneOrMoreKeys()
 
             if (_queuedKeys.Count == 0)
             {
-                var key = _console.ReadKey();
-                _lastNKeys.Enqueue(key);
-                _queuedKeys.Enqueue(key);
+                while (!_charMap.KeyAvailable)
+                {
+                    // Don't want to block when there is an escape sequence being read.
+                    if (_charMap.InEscapeSequence)
+                    {
+                        if (_console.KeyAvailable)
+                        {
+                            _charMap.ProcessKey(_console.ReadKey());
+                        }
+                        else
+                        {
+                            // This should be ok since it will only run if there
+                            // are no keys waiting to be read.
+                            Thread.Sleep(5);
```

Comment:
Should the sleep happen in `_charMap` which knows the maximum timeout after escape?

Reply:
The idea behind this is that as soon as it sees something that isn't part of an escape sequence, there's no need to wait out the whole timer, so this is more of a CPU yield during a poll loop than a sleep. I'll add a comment explaining that.